src/middleware/error-handler.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCodes } from "http-status-codes";
+
+// Define a custom error type if needed
+interface CustomError extends Error {
+  statusCode?: number;
+  code?: number;
+  errors?: Record<string, { message: string }>;
+  keyValue?: Record<string, any>;
+  value?: string;
+}
+
+export const errorHandlerMiddleware = (
+  err: CustomError, // Ensure this is typed as a CustomError
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void => {
+  let customError = {
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || "Something went wrong, try again later",
+  };
+
+  if (err.name === "ValidationError") {
+    customError.msg = Object.values(err.errors!)
```

Comment:
Why do we know `err.errors` is defined if this is a ValidationError? I assume there is some documentation somewhere that says "if we produce a ValidationError, err.errors will be a valid array" but I don't know this as someone reading this code. Link the documentation here!